scripts/tests/test_compare_implementations.py

```
@@ -0,0 +1,318 @@
+    def test_reanalysis_requires_the_complete_canonical_artifact_inventory(self) -> None:
+        """Removing manifest records cannot bypass retained-artifact verification."""
+        with TemporaryDirectory() as temporary_directory:
+            bundle = Path(temporary_directory) / "comparison"
+            run_comparison(bundle, _emitter("cpp"), _emitter("rust"), _sources(), 10)
+            manifest_path = bundle / "manifest.json"
+            manifest = load_json(manifest_path)
+            manifest["artifacts"] = []
+            write_json(manifest_path, manifest)
+            (bundle / "raw" / "rust" / "stderr.txt").unlink()
+
+            with self.assertRaisesRegex(ValueError, "artifact inventory is incomplete"):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)